packages/flutter_tools/test/general.shard/asset_bundle_test.dart

```diff
@@ -361,180 +335,99 @@ flutter:
       Platform: () => platform,
       ProcessManager: () => FakeProcessManager.any(),
     });
+  });
 
-    group('flavors feature', () {
-      Future<ManifestAssetBundle> buildBundleWithFlavor(String? flavor) async {
-        final ManifestAssetBundle bundle = ManifestAssetBundle(
-          logger: logger,
-          fileSystem: testFileSystem,
-          platform: platform,
-          flutterRoot: Cache.defaultFlutterRoot(
-            platform: platform,
-            fileSystem: testFileSystem,
-            userMessages: UserMessages(),
-          ),
-          splitDeferredAssets: true,
-        );
-
-        await bundle.build(
-          packagesPath: '.packages',
-          flutterProject: FlutterProject.fromDirectoryTest(testFileSystem.currentDirectory),
-          flavor: flavor,
-        );
-        return bundle;
-      }
+  group('AssetBundle.build (hermetic)', () {
+    testUsingContext('throws ToolExit when directory entry contains invalid characters (Windows only)', () async {
```

Comment:
What globals are we referencing from the context, and should we explicitly override them?

Reply:
I ask because these types of tests make it easy to write tests that look like we're creating the appropriate fakes, while the test under code silently uses real globals it got from the context.

Reply:
This was a mistake. I forgot to change this to `testWithoutContext`. Working on it.

Reply:
Addressed.